rpisec/rpis_camera.py

```diff
@@ -146,17 +123,90 @@ def set_motion_settings(self):
         self.camera.awb_gains = awb_gains
         self.camera.exposure_mode = 'off'
 
-    def start_motion_detection(self):
-        try:
-            if self.camera.recording:
-                self.camera.wait_recording(0.1)
+    def start_motion_detection(self, rpis):
+        min_area = 500
+        past_frame = None
+        logger.debug("Started motion detection from video stream from RpiCamera")
+        # loop over the frames of the video
+        picture_path = '/tmp/rpi-security-current.jpg'
+        while not self.lock.locked() and rpis.state.current == 'armed':
+            self.camera.resolution = self.motion_size
+            self.camera.capture(picture_path, use_video_port=False)
+            time.sleep(0.3)
+            # grab the current frame
+            frame = cv2.imread(picture_path)
+
+            # if frame is initialized, we have not reach the end of the video
+            if frame is not None:
+                past_frame = self.handle_new_frame(frame, past_frame, min_area)
             else:
-                logger.debug("Starting motion detection")
-                self.set_motion_settings()
-                self.motion_detector.motion_detection_started = time.time()
-                self.camera.start_recording(os.devnull, format='h264', motion_output=self.motion_detector)
-        except Exception as e:
-            logger.error('Error in start_motion_detection: {0}'.format(repr(e)))
+                logger.error("No more frame")
+            rpis.state.check()
+        else:
+            self.stop_motion_detection()
+
+    def handle_new_frame(self, frame, past_frame, min_area):
+        (h, w) = frame.shape[:2]
+        r = 500 / float(w)
+        dim = (500, int(h * r))
+        frame = cv2.resize(frame, dim, cv2.INTER_AREA) # We resize the frame
```

Comment:
Hey @qlerebours, random question, why did you resize the frame here? Is it to save on resources?

Reply:
I don't remember exactly why I did that, I think it was to save on resources, but I'm not sure because it's been a few monthes since I developed it :-/
Does it cause troubles ?

Reply:
No worries. I was just going to increase it to make the photo with the bounding box larger but wasn't sure why you chose 500 pixels.